metrics/ci_integrations/test/destination/firestore/adapter/firestore_destination_client_adapter_test.dart

```diff
@@ -79,20 +79,21 @@ void main() {
     );
 
     test(
-      ".addBuilds() returns normally if firestore throws GrpcError.notFound",
+      ".addBuilds() throws an exception if the firestore throws exception different from GrpcError.notFound",
```

Comment:
```suggestion
      ".addBuilds() rethrows an exception thrown by the firestore",
```
Not sure this test makes sense since the current implementation rethrows `GrpcError` and does not catch other errors. The only way I see is to introduce a custom exception here and don't rethrow `GrpcError`. Thus, we can ensure in tests that implementation catches all `GrpcError`s and wraps it in a custom. I suppose this makes sense as makes us independent from the low-level package-related exceptions. What do you think? 

cc @illia-romanenko  @solid-vovabeloded

Reply:
I agree with @solid-vladlaba - excellent suggestion!